docs/walkthrough/updates.rst

```diff
@@ -26,9 +26,8 @@ with one additional line:
     :emphasize-lines: 24
 
     @kopf.on.create('zalando.org', 'v1', 'ephemeralvolumeclaims')
-    def create_fn(body, spec, meta, namespace, logger, **kwargs):
+    def create_fn(spec, name, namespace, logger, **kwargs):
 
-        name = meta.get('name')
         size = spec.get('size')
         if not size:
```

Comment:
`if not spec.get('size'):`  in the best case.

For that specific case, `size` is reused in 3 places — so, it is made a variable.

This approach — making `spec.size` notation possible — will make some use-cases misaligned: `body.metadata.labels` is a live-view mapping even if the body is replaced, while `body.spec.struct` is that dictionary itself, and is not updated with the body is replaced (because it is not a well-known field).

It is even more complicated for the cases with 2+ levels of nestedness:

```yaml
spec:
  struct:
    substruct:
      field: value
```

In this example, should `spec.struct` be a pure `dict`, or a live-view mapping? In the former case, `spec.struct.substruct` is not possible. In the latter case, how do we know whether `spec.children` (nonexistent here) should be a mapping or a list? Is it possible to make `spec.children.keys()` or `for val in spec.children`? Or maybe it is a number?

All in all, it is difficult to implement live views for arbitrary types: mappings, collections, scalars. Especially scalars. And especially when the fields can be absent and we have no schema.

For these reasons, I decided to wrap only few well-known fields and expose them as attributes/properties (dot-notation), and keep the rest accessable as keys-values (dict-notation).

--- 
**PS:** CRDs have schemas (optionally). But if we add schema support with dynamic DSL sugar, Kopf will turn into a full-fledged K8s/OpenAPI client library, which I try to avoid. 

I'd better make a config for wrapping RawBody values into arbitrary classes, and let the developers decide how they want to get it to their handlers, and let someone else develop their own DSL-enabling wrappers.
 
Or, having such a library, they can do it right now with an extra line like `body = pykube.Pod(api, body); print(body.name)` (probably).

Reply:
I think I mostly understand what you're getting at. :) I didn't realize that this live-view mechanism only works for some fields, but now with your explanation it makes sense.